protocol/SignalingProtocol.proto

```diff
@@ -56,12 +56,14 @@ enum SdkJoinFlags {
 }
 
 message SdkClientDetails {
-    required string app_version_name = 1;
-    required string app_version_code = 2;
-    required string device_model = 3;
-    required string device_make = 4;
-    required string platform_name = 5;
-    required string platform_version = 6;
+    optional string app_version_name = 1;
+    optional string app_version_code = 2;
+    optional string device_model = 3;
+    optional string device_make = 4;
+    optional string platform_name = 5;
+    optional string platform_version = 6;
+    optional string client_source = 7;
```

Comment:
What are these two fields for? They are not related to the dropped field right?

Reply:
These are new fields that the server accepts. This PR updates the signaling protocol to match the server definition.